### PR TITLE
GH-148 Splits chunks on whitespace or punctuation

### LINKED ISSUE #148: Split long lines on whitespace
Given response body is usually a plain text
And log is intended to be read by human
When logbook splits long line into chunks
Then split boundary should be (possibly) located at white-space, within say size ± delta interval

1. Split boundary should be within `[size-delta, size]` so chunk is not longer then `size`
2. `delta` should be fixed size of e.g. to average "token" length. I think 10 to 16 characters should be fine
3. Split should happen on whitespaces as well as after common json punctuation characters `,:`

Having said that split algorithm could leap for `size` number of characters and look back `delta` characters for whitespaces and punctuation.

Imporovement on top of #142

### PR DESCRIPTION
**Pull request by AlexanderYastrebov**
Fixes #148

### CODE REVIEW

File: logbook-core/src/main/java/org/zalando/logbook/StringSpliterator.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
+package org.zalando.logbook;
+
+import java.util.Spliterator;
+import java.util.function.Consumer;
+
+class StringSpliterator implements Spliterator<String> {
```
Comment (AlexanderYastrebov) [inline logbook-core/src/main/java/org/zalando/logbook/StringSpliterator.java]: Should it be renamed to `Chunk(ing)Spliterator`?

Comment (whiskeysierra) [inline logbook-core/src/main/java/org/zalando/logbook/StringSpliterator.java]: I like `ChunkingSpliterator`.

File: logbook-core/src/main/java/org/zalando/logbook/StringSpliterator.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            final char ch = string.charAt(i - 1);
+            if (isSplitCharacter(ch)) {
+                return string.substring(position, i);
+            }
+        }
+        return string.substring(position, chunkMax);
+    }
+
+    private static boolean isSplitCharacter(final char ch) {
+        return ch == ' ' || ch == ',' || ch == ':';
```
Comment (whiskeysierra) [inline logbook-core/src/main/java/org/zalando/logbook/StringSpliterator.java]: Could be a switch.

File: logbook-core/src/test/java/org/zalando/logbook/StringSpliteratorTest.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    }
+
+    @Test
+    public void shouldNotSupportPartitions() {
+        assertThat(new StringSpliterator("", 1).trySplit(), is(nullValue()));
+    }
+
+    @Test
+    public void shouldBeSizedWhenMinEqualToMax() {
+        assertTrue((new StringSpliterator("Hello", 5).characteristics() | SIZED) != 0);
```
Comment (whiskeysierra) [inline logbook-core/src/test/java/org/zalando/logbook/StringSpliteratorTest.java]: There should probably be the inverse test for this as well.

**Comment by whiskeysierra** [comment]
:+1:

**Comment by whiskeysierra** [comment]
:+1:

**Pull request merged**